### DISCUSSION TITLE
Understanding 'memory' (storageType) in storageType^storagePath#property

### DISCUSSION THREAD

**Discussion by dmantisk** [mProjectsCode/obsidian-meta-bind-plugin#469] [Q&A] [answered]
I'll explain my confusion using the following examples all of which creates a simple list with selection

#### Example 1
```meta-bind
INPUT[select(
option(option a),
option(option b)
)]
```
This does NOT store the selection anywhere; It works exactly as expected
#### Example 2
```meta-bind
INPUT[select(
option(option a),
option(option b)
):memory]
```
This creates a property named 'memory' and stores the data in that; I expected it to not create any properties, but temporarily store the data in the file (I don't know where exactly)
#### Example 3
```meta-bind
INPUT[select(
option(option a),
option(option b)
):memory^gibberish]
```
This temporarily stores the data somewhere WITHOUT creating a property. I added 'gibberish' because what I type there doesn't seem to matter; This is the behaviour I expected from example 2.

### Question
Is this how it's supposed to work?

And if so where exactly is the temporary data stored for example 3? I want to be sure that the data is only temporary and doesn't end up accumulating somewhere

**Comment by mProjectsCode** [comment]
Yes, this is as expected. [This](https://www.moritzjung.dev/obsidian-meta-bind-plugin-docs/guides/bindtargets/) docs page tries to explain it. In your second example, you only specify a property, so the storage type defaults to frontmatter.

The `memory` storage type stores the values in memory, meaning they do not persist anywhere.

**Reply by mProjectsCode** [reply, marked as answer]
The reason you need the "gibberish" property is so that you can have multiple independent memory properties in the same file and you can use the property to reference it elsewhere. E.g. a `VIEW[{memory^gibberish}]`.

**Reply by dmantisk** [reply]
Thank you.
It being needed as an ID cleared my confusion.

Related to that, is it possible to set up if conditions for attributes like class? (Without js)
For example, have a VIEW field use one or the other class depending on the data in memory?
(Should I make a new post for this?)

**Reply by mProjectsCode** [reply]
If you mean CSS classes, then no, not without JS.

**Reply by dmantisk** [reply]
That's disheartening, but thank you
